src/components/DoctorCard/DoctorCard.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useState } from "react";
+import { Button, Collapse } from "@material-ui/core";
+import DoctorCardStyle from "./DoctorCard.style";
+import { IDoctorCard } from "./DoctorCard.interface";
+import { useRequest } from "hooks/useRequest";
+import { AddressDetails } from "components/AddressDetails";
+import { DoctorAvatar } from "components/DoctorAvatar";
+import { DoctorAvailability } from "components/DoctorAvailability";
+
+const DoctorCard: React.FC<IDoctorCard> = (props) => {
+  const { id, address, name } = props;
+  const classes = DoctorCardStyle();
+  const [open, setOpen] = useState(false);
+  const { apiData, isLoading, execute } = useRequest("GET", `availabilities?doctorId=${id}`, false);
+
+  const handleClick = () => {
+    setOpen(!open);
```

Comment:
When setting a state using the previous value, it is better to use [functional update](https://reactjs.org/docs/hooks-reference.html#functional-updates)
```suggestion
    setOpen(currentOpen => !currentOpen);
```

Reply:
I know that way, but why this is better? what vantage do we have?